controllers/medusa/medusarestorejob_controller.go

```diff
@@ -244,44 +238,27 @@ func (r *MedusaRestoreJobReconciler) podTemplateSpecUpdateComplete(ctx context.C
 	return true, nil
 }
 
-func (r *MedusaRestoreJobReconciler) prepareRestore(ctx context.Context, request *medusa.RestoreRequest) (bool, error) {
-	// Create a prepare_restore medusa task to create the mapping files in each pod.
-	// Returns true if the reconcile needs to be requeued, false otherwise.
-	prepare := &medusav1alpha1.MedusaTask{}
-	if err := r.Client.Get(context.Background(), types.NamespacedName{Name: request.RestoreJob.Status.RestoreKey, Namespace: request.RestoreJob.Namespace}, prepare); err != nil {
-		if errors.IsNotFound(err) {
-			// Create the sync task
-			prepare = &medusav1alpha1.MedusaTask{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      request.RestoreJob.Status.RestoreKey,
-					Namespace: request.RestoreJob.Namespace,
-				},
-				Spec: medusav1alpha1.MedusaTaskSpec{
-					Operation:           medusav1alpha1.OperationTypePrepareRestore,
-					CassandraDatacenter: request.RestoreJob.Spec.CassandraDatacenter,
-					BackupName:          request.MedusaBackup.ObjectMeta.Name,
-					RestoreKey:          request.RestoreJob.Status.RestoreKey,
-				},
-			}
-			if err := r.Client.Create(context.Background(), prepare); err != nil {
-				return true, err
-			}
-		} else {
-			return true, err
-		}
-	} else {
-		if !prepare.Status.FinishTime.IsZero() {
-			// Prepare is finished
-			return false, nil
-		}
-		if len(prepare.Status.InProgress) == 0 {
-			// No more pods are running the task but finish time is not set.
-			// This means the task failed.
-			return false, fmt.Errorf("prepare restore task failed for restore %s", request.RestoreJob.Name)
-		}
+// prepareRestore prepares the MedusaRestoreMapping for the restore operation.
+// It uses the Medusa client to get the host map for the restore operation, using the Medusa standalone service to get the backup topology.
+func (r *MedusaRestoreJobReconciler) prepareRestore(ctx context.Context, request *medusa.RestoreRequest) (*medusav1alpha1.MedusaRestoreMapping, error) {
+	medusaClient, err := r.ClientFactory.NewClient(medusaServiceUrl(
+		request.Datacenter.Spec.ClusterName,
```

Comment:
Observation: We really need to start distinguishing between the ClusterName for the k8s resource and the Cassandra cluster name. Impossible to tell what is going on throughout these codebases sometimes :/

Reply:
> The only way to get the cluster name, which is used to build the service url, is .spec.ClusterName here.

Do you mean .spec.cassandra.clusterName? This can be set differently to the k8s resource name and doesn't need to bear any relation to it AFAIK.

I believe you could possibly use .metadata.labels to get the parent cluster name.

Reply:
Conversation resolved offline: we don't want to use the k8c resource name here, we really need the sanitized name.